crates/holochain_p2p/src/spawn/actor.rs

```diff
@@ -1256,18 +1269,21 @@ impl actor::HcP2p for HolochainP2pActor {
 
             let op_hash_list: Vec<OpId> = op_hash_list.into_iter().map(|h| h.to_k2_op()).collect();
 
-            let urls: std::collections::HashSet<Url> = space
-                .peer_store()
-                .get_near_location(basis_hash.get_loc(), usize::MAX)
-                .await?
-                .into_iter()
-                .filter_map(|info| {
-                    if info.is_tombstone {
-                        return None;
-                    }
-                    info.url.clone()
-                })
-                .collect();
+            let urls: std::collections::HashSet<Url> = get_remote_agents_near_location(
```

Comment:
We weren't being careful about local agents on publish. When you remember that the publish workflows splits up publish requests by basis hash, then this code would run multiple times and always find ourselves close to the location for agent activity ops. This would pretty much always hold up publish. That's a good thing because it explains the poor UX in yesterday's testing session.